quarkus/service/src/main/java/org/apache/polaris/service/quarkus/metrics/RealmIdTagContributor.java

```diff
@@ -39,15 +40,25 @@ public Tags contribute(Context context) {
     HttpServerRequest request = context.request();
     try {
       RealmContext realmContext = resolveRealmContext(request);
-      return Tags.of(TAG_REALM, realmContext.getRealmIdentifier());
+      return realmContext == null
+          ? Tags.empty()
+          : Tags.of(TAG_REALM, realmContext.getRealmIdentifier());
     } catch (Exception ignored) {
       // ignore, the RealmContextFilter will handle the error
       return Tags.empty();
     }
   }
 
+  @Nullable
   private RealmContext resolveRealmContext(HttpServerRequest request) {
-    return realmContextResolver.resolveRealmContext(
-        request.absoluteURI(), request.method().name(), request.path(), request.headers()::get);
+    return realmContextResolver
+        .resolveRealmContext(
+            request.absoluteURI(), request.method().name(), request.path(), request.headers()::get)
+        .exceptionally(error -> null)
+        .toCompletableFuture()
+        // get the result of the CompletableFuture if it's already completed,
+        // otherwise return null as this code is executed on an event loop thread,
+        // and we don't want to block it.
```

Comment:
[doubt] This might be a naive question, are we implying since it's not resolved we will treat this as null ? what if my resolution is slow, in short are we saying if by the time we reach here if the realm is not resolved then return null and hence fail ?